src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+
+    public Figure getRandomFigure() {
+        GeometricFigure figure = new GeometricFigureSupplier().getRandomFigure();
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Random random = new Random();
+        if (figure == GeometricFigure.CIRCLE) {
+            return new Circle(random.nextInt(), colorSupplier.getRandomColor().toLowerCase());
+        }
+        if (figure == GeometricFigure.RECTANGLE) {
+            return new Rectangle(random.nextInt(), random.nextInt(),
+                    colorSupplier.getRandomColor().toLowerCase());
+        }
+        if (figure == GeometricFigure.ISOSCELES_TRAPEZOID) {
+            return new IsoscelesTrapezoid(random.nextInt(), random.nextInt(),
+                    random.nextInt(), colorSupplier.getRandomColor().toLowerCase());
+        }
+        if (figure == GeometricFigure.SQUARE) {
+            return new Square(random.nextInt(), colorSupplier.getRandomColor().toLowerCase());
+        }
+        return new RightTriangle(random.nextInt(), random.nextInt(),
+                colorSupplier.getRandomColor().toLowerCase());
```

Comment:
All magic numbers in your code should be constants. When calling 'nextInt()', you should specify a bound to avoid getting negative or too large values. Also, don't pass expressions in the constructor. Create separate variables and pass them on for better code readability.